aws/s3_test.go

```diff
@@ -495,26 +497,67 @@ func TestFilterS3Bucket_Config(t *testing.T) {
 		require.NoErrorf(t, err, "Failed to create test bucket - %s", bucketName)
 	}
 
+	awsSession, err := S3TestCreateNewAWSSession("")
+	require.NoError(t, err, "Failed to create random session")
+
 	// Clean up test buckets
-	defer nukeAllS3Buckets(awsParams.awsSession, aws.StringSlice(bucketNames), 1000)
+	// defer nukeAllS3Buckets(awsParams.awsSession, aws.StringSlice(bucketNames), 1000)
```

Comment:
@brikis98 With this line commented, no buckets are deleted, the test cases in here all pass.

With it uncommented, the buckets are deleted, here's the output:

![image](https://user-images.githubusercontent.com/13165182/84520950-edee1b80-ac99-11ea-9ff0-b51d8963672b.png)

Reply:
Moving defer into the `t.Run` block seems to work.